lib/src/value/color.ts

```diff
@@ -3,371 +3,1112 @@
+      throw valueError(
+        `Expected \`weight\` between \`0\` and \`1\`; received \`${weight}\`.`
+      );
     }
+
+    // ColorJS inverses the `weight` argument, where `0` is `this` and `1` is
+    // `color2`.
+    const color = this.color.mix(color2.color, 1 - weight, {
+      space: encodeSpaceForColorJs(this.space),
+      hue: hueInterpolationMethod,
+    } as any); // @TODO Waiting on new ColorJS release to fix type defs
+    const coords = getCoordsFromColor(color.coords, this.space === 'rgb');
+    return new SassColor({
+      space: this.space,
+      [this.channel0Id]: coords[0],
+      [this.channel1Id]: coords[1],
+      [this.channel2Id]: coords[2],
+      alpha: NaNtoNull(this.color.alpha),
+    });
   }
 
-  // Computes `this`'s red`, `green`, and `blue` channels based on `hue`,
-  // `saturation`, and `value`.
-  //
-  // Algorithm from the CSS3 spec: https://www.w3.org/TR/css3-color/#hsl-color.
-  private hslToRgb(): void {
-    const scaledHue = this.hue / 360;
-    const scaledSaturation = this.saturation / 100;
-    const scaledLightness = this.lightness / 100;
+  /**
+   * Returns a new color that's the result of changing one or more of this
+   * color's channels.
+   */
+  change(
+    options: {
+      [key in ChannelNameHsl]?: number | null;
+    } & {
+      space?: ColorSpaceHsl;
+    }
+  ): SassColor;
+  change(
+    options: {
+      [key in ChannelNameHwb]?: number | null;
+    } & {
+      space?: ColorSpaceHwb;
+    }
+  ): SassColor;
+  change(
+    options: {
+      [key in ChannelNameLab]?: number | null;
+    } & {
+      space?: ColorSpaceLab;
+    }
+  ): SassColor;
+  change(
+    options: {
+      [key in ChannelNameLch]?: number | null;
+    } & {
+      space?: ColorSpaceLch;
+    }
+  ): SassColor;
+  change(
+    options: {
+      [key in ChannelNameRgb]?: number | null;
+    } & {
+      space?: ColorSpaceRgb;
+    }
+  ): SassColor;
+  change(
+    options: {
+      [key in ChannelNameXyz]?: number | null;
+    } & {
+      space?: ColorSpaceXyz;
+    }
+  ): SassColor;
+  change(
+    options: {
+      [key in ChannelName]?: number | null;
+    } & {
+      space?: KnownColorSpace;
+    }
+  ) {
+    const spaceSetExplicitly = !!options.space;
+    let space = options.space ?? this.space;
+    if (this.isLegacy && !spaceSetExplicitly) {
+      if (
+        propertyIsSet(options.whiteness) ||
+        propertyIsSet(options.blackness) ||
+        (this.space === 'hwb' && propertyIsSet(options.hue))
+      ) {
+        space = 'hwb';
+      } else if (
+        propertyIsSet(options.hue) ||
+        propertyIsSet(options.saturation) ||
+        propertyIsSet(options.lightness)
+      ) {
+        space = 'hsl';
+      } else if (
+        propertyIsSet(options.red) ||
+        propertyIsSet(options.green) ||
+        propertyIsSet(options.blue)
+      ) {
+        space = 'rgb';
+      }
+      if (space !== this.space) {
+        emitColor4ApiChangeSpaceDeprecation();
+      }
+    }
```

Comment:
Can we factor this out into a helper?

Reply:
Done in https://github.com/sass/embedded-host-node/pull/259/commits/18ab10ee84ce3b3321b8a8b98f52c7fb79650653